packages/client/src/runtime/RequestHandler.ts

```diff
@@ -32,17 +32,30 @@ export type RequestParams = {
   engineHook?: EngineMiddleware
   args: any
   headers?: Record<string, string>
-  transactionId?: string
+  transactionId?: string | number
   unpacker?: Unpacker
 }
 
 export type Request = {
   document: Document
   runInTransaction?: boolean
-  transactionId?: string
+  transactionId?: string | number
   headers?: Record<string, string>
 }
 
+function getRequestInfo(requests: Request[]) {
+  const txId = requests[0].transactionId
+  const inTx = requests[0].runInTransaction
+  const headers = requests[0].headers
+
+  // if the tx has a number for an id, then it's a regular batch tx
+  const _inTx = typeof txId === 'number' && inTx ? true : undefined
+  // if the tx has a string for id, it's an interactive transaction
+  const _txId = typeof txId === 'string' && inTx ? txId : undefined
+
+  return { inTx: _inTx, headers: { transactionId: _txId, ...headers } }
+}
```

Comment:
How is it a string and not a number for interactive transaction?

Reply:
It's a GUID

Reply:
So when interactive transaction go GA, would we have GUIDs everywhere? (and not number like current $transaction GA)

Reply:
So I restored the behavior, tx ids are `string` and continue going through `then` and regular txs have `number` ids and go through `requestTransaction`.